v0/destinations/am/transform.js

```diff
@@ -516,17 +516,75 @@ function trackRevenueEvent(message, destination) {
     destination.Config.trackRevenuePerProduct === true ||
     destination.Config.trackProductsOnce === false
   ) {
-    const productPurchasedEvents = getProductPurchasedEvents(
-      message,
-      destination
-    );
-    if (productPurchasedEvents.length > 0) {
-      sendEvents = [...sendEvents, ...productPurchasedEvents];
-    }
+    sendEvents = getProductPurchasedEvents(message, destination, sendEvents);
   }
   return sendEvents;
 }
 
+function updateTraitsObject(property, traitObject, newProperty) {
+  let propertyToUpdate;
+  if (traitObject.hasOwnProperty(property)) {
+    propertyToUpdate = traitObject[property];
+    newProperty[property] = propertyToUpdate;
+    delete traitObject[property];
+  }
+  return traitObject;
+}
+
+function passingTraitConfigs(
+  configPropertyTrait,
+  objectToInclude,
+  traitObject
+) {
+  let updatedTrait;
+  configPropertyTrait.forEach(traitCount => {
+    const property = traitCount.traits;
+    updatedTrait = updateTraitsObject(property, traitObject, objectToInclude);
+  });
+  return updatedTrait;
+}
+
+function traitsHandler(message, destination) {
+  let updatedTrait;
+  const messageBuffer = JSON.parse(JSON.stringify(message));
+  const traitObject = JSON.parse(JSON.stringify(messageBuffer.context.traits));
+
+  if (destination.Config.traitsToIncrement) {
+    traitObject.$add = new Object();
+    updatedTrait = passingTraitConfigs(
+      destination.Config.traitsToIncrement,
+      traitObject.$add,
+      traitObject
+    );
+  }
+  if (destination.Config.traitsToSetOnce) {
+    traitObject.$setOnce = new Object();
+    updatedTrait = passingTraitConfigs(
+      destination.Config.traitsToSetOnce,
+      traitObject.$setOnce,
+      traitObject
```

Comment:
can we pass only `traitObject` and the other parameter as string like, `$setOnce` or `$add` and prepare the same in called method?